src/osd/OSD.cc

```diff
@@ -4008,6 +4008,23 @@ void OSD::load_pgs()
     // read pg state, log
     pg->read_state(store);
 
+    if (pg->dne())  {
+      dout(10) << "load_pgs " << *it << " deleting dne" << dendl;
+      pg->ch = nullptr;
+      service.pg_remove_epoch(pg->pg_id);
+      pg->unlock();
+      // Delete pg
+      RWLock::WLocker l(pg_map_lock);
```

Comment:
I'm not sure it matters at this point in the boot process, but we need to drop this before invoking recursive_remove_collection() since that prompts disk accesses. I'd just put a block around the pg_map modifying bits.